File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i in range(len(nums)):
            rem = target - nums[i]
            if rem in map:
                return [i, map[rem]]
            else:
                map[nums[i]] = i
            
        return [-1,-1]

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hmap = new HashMap<>();

        for(int i=0; i<nums.length; i++){
            int remainder = target - nums[i];
            if(hmap.containsKey(remainder))
                return new int[] {i, hmap.get(remainder)};

            hmap.put(nums[i], i);
        }

        return new int[] {-1, -1};
    }
}